File: LeetCode/JavaScript/Arrays/moveZeroes.js
//beat 100%
var moveZeroes = function(nums) {
    var num_spliced = 0;
    for(var i = 0; i< nums.length; i++){      
        if(nums[i]==0){
            nums.splice(i,1);  
            num_spliced++;
            i--;
        }
    }
    for(var j = 0; j < num_spliced; j++){
        nums.push(0);   
    }
};

File: LeetCode/Java/Arrays/MoveZeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int fast = 1;
        int slow = 0;
        int tmp;
        while(fast < nums.length){
            if(fast == slow){
                fast++;
            }
            if(nums[fast] == 0){
                fast++;
            }
            else if(nums[slow] != 0){
                slow++;
            }
            else{
                tmp = nums[fast];
                nums[fast] = nums[slow];
                nums[slow] = tmp;
            }
        }
    }
}